cime_config/machines/cmake_macros/crayclang-scream_frontier-scream-gpu.cmake

```diff
@@ -1,6 +1,6 @@
 set(MPICC "mpicc")
-set(MPICXX "hipcc")
-set(MPIFC "ftn")
+set(MPICXX "mpicxx") # Needs MPICH_CXX to use hipcc
```

Comment:
Need to make sure mpicxx is picking up the right compiler (hipcc in this case). To avoid this, we previously had hipcc and then added MPI include and lib flags appropriately.

Reply:
To elaborate, hipcc will pick it up from the amd/rocm modules while Cray wrapper (CC) will use the Cray's C++ compiler. Depending on what the path ends up being, mpicxx might pickup the Cray's compiler which lags behind the AMD's LLVM featureset/

Reply:
The env var `MPICH_CXX` should take care of this. See the changes in the config_machines.xml section.

Reply:
>       <env name="MPICH_CXX">$SHELL{which hipcc}</env>

Clarify the purpose/usage of MPICXX and MPICH_CXX. Trying to understand why we need both of them and if there is way to improve clarity.

Reply:
`MPICXX` is a CMake var, used by E3SM CMake build system to specify the compiler. We could do away with this, and just do `set(CMAKE_CXX_COMPILER mpicxx CACHE STRING "")`, or something like that (maybe with FORCE), but I think for bwd reasons we need to keep these vars around for CIME to pick up the compiler (that's a guess). `MPICH_CXX`, otoh, is an env var that it used by `mpicxx` to decide which backend CXX compiler to use.

Reply:
Yep, same concept as OMPI_CXX=nvcc for openmpi on cuda machines.